File: apps/myjobhunter/backend/app/schemas/application/document.py
"""Document schemas — Phase 2."""
import uuid
from datetime import datetime

from pydantic import BaseModel


class DocumentRead(BaseModel):
    id: uuid.UUID
    application_id: uuid.UUID | None = None
    title: str
    kind: str
    body: str | None = None
    file_path: str | None = None
    filename: str | None = None
    content_type: str | None = None
    size_bytes: int | None = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: apps/myjobhunter/frontend/src/types/document/document.ts
import type { DocumentKind } from "@/types/document/document-kind";

export interface Document {
  id: string;
  user_id: string;
  application_id: string | null;
  title: string;
  kind: DocumentKind;
  body: string | null;
  filename: string | null;
  content_type: string | null;
  size_bytes: number | null;
  has_file: boolean;
  deleted_at: string | null;
  created_at: string;
  updated_at: string;
}
